tests/phpunit/tests/block-library/commentTemplate.php

```diff
@@ -0,0 +1,206 @@
+<?php
+/**
+ * Comment Template block tests
+ *
+ * @package WordPress
+ * @subpackage Block Library
+ * @since 6.0.0
+ */
+
+/**
+ * Tests for the Comment Template block.
+ *
+ * @since 6.0.0
+ *
+ * @group block-library
+ */
+class Test_Block_Library_CommentTemplate extends WP_UnitTestCase {
+
+	private static $custom_post;
+	private static $comment_ids;
+	private static $per_page = 5;
+
+	public function set_up() {
+		parent::set_up();
+
+		update_option( 'page_comments', true );
+		update_option( 'comments_per_page', self::$per_page );
+		update_option( 'comment_order', 'ASC' );
+
+		self::$custom_post = self::factory()->post->create_and_get(
+			array(
+				'post_type'    => 'dogs',
+				'post_status'  => 'publish',
+				'post_name'    => 'metaldog',
+				'post_title'   => 'Metal Dog',
+				'post_content' => 'Metal Dog content',
+				'post_excerpt' => 'Metal Dog',
+			)
+		);
+
+		self::$comment_ids = self::factory()->comment->create_post_comments(
+			self::$custom_post->ID,
+			1,
+			array(
+				'comment_author'       => 'Test',
+				'comment_author_email' => 'test@example.org',
+				'comment_content'      => 'Hello world',
+			)
+		);
+	}
+
+	function test_build_comment_query_vars_from_block_with_context() {
```

Comment:
I noticed this test is failing in my local environment. I'm taking a look into that. 👀 

```
1) Test_Block_Library_CommentTemplate::test_build_comment_query_vars_from_block_with_context
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
     'status' => 'approve'
     'no_found_rows' => false
     'update_comment_meta_cache' => false
-    'post_id' => 4
     'hierarchical' => 'threaded'
     'number' => 5
     'paged' => 1
 )
```

Reply:
I see the same error locally. It's really surprising.

Reply:
For some reason, the block is not being instantiated here with context; it's empty.

https://github.com/WordPress/wordpress-develop/blob/537d27225db483c6c16afc1daef6364d4f7668e4/tests/phpunit/tests/block-library/commentTemplate.php#L57-L62

I see the same happening in all the tests where context is used. Could it be related to the block not included in WordPress core yet? 🤔

Reply:
I think I know why, we need to have the block code first and we can then land the tests. All good then. Let's wait for https://github.com/WordPress/wordpress-develop/pull/2564 and see how it looks afterwards. Good detective work 😄